test/unit/spec/EnrollQuestions_spec.js

```diff
@@ -25,23 +25,24 @@ function (Q, Okta, OktaAuth, Util, Form, Beacon, Expect, Router, RouterUtil, Bro
 
   var { _, $ } = Okta;
   var itp = Expect.itp;
-  var tick = Expect.tick;
 
   function setup (res, startRouter, languagesResponse) {
     var setNextResponse = Util.mockAjax();
     var baseUrl = 'https://foo.com';
     var authClient = new OktaAuth({ url: baseUrl, transformErrorXHR: LoginUtil.transformErrorXHR });
     var afterErrorHandler = jasmine.createSpy('afterErrorHandler');
+    var successSpy = jasmine.createSpy('success');
     var router = new Router({
       el: $sandbox,
       baseUrl: baseUrl,
       authClient: authClient,
+      globalSuccessFn: successSpy,
       'features.router': startRouter
     });
     router.on('afterError', afterErrorHandler);
     Util.registerRouter(router);
     Util.mockRouterNavigate(router, startRouter);
-    return tick()
+    return Q()
```

Comment:
same question as in the other test, if we can return the logic from the `then` block below

Reply:
turns out I cannot because the whole chain is like
```
setup() {
  return Util.mockIntrospectResponse.then(..).then(..)
}
.then(..)
.then(* clear up tests; done(); */)                  <-- from Expect.runTest
.done(done)                                          <-- from Expect.runTest
```

When `Util.mockIntrospectResponse` is resolved, it will trigger `done(done)` when ends test hence cause many failures. Not sure if I did some other stupid thing but this was what I found last time.

Reply:
oh.... I see, a little more complicated then... sigh... thanks for trying...
Maybe it needs to load the page first, and then it can do the introspect call... so this `Q` could be a `Expect.waitForPrimaryAuth` or something like that, since we can't really skip this step. We need to load the initial page, then introspect, which refreshes auth-state, and then go on... 

Anyways, if you want to try it, go for it, but it is good as is, already better than `tick`

Reply:
Sounds good. Will give an try next time I visit this file ;)